lib/ask/runtime/session.ex

```diff
@@ -79,9 +79,10 @@ defmodule Ask.Runtime.Session do
         # do not retry since the respondent was never contacted, thus the retries should not be consumed
         session = contact_respondent(session, runtime_channel)
 
+        now = DateTime.utc_now
         next_available_date_time = session.schedule |> Schedule.next_available_date_time
 
-        base_timeout = Interval.new(from: DateTime.utc_now, until: next_available_date_time)
+        base_timeout = Interval.new(from: now, until: next_available_date_time)
```

Comment:
How about leaving `DateTime.utc_now` ? Are we storing it in a variable just for improving readability a bit or for any other reason I might be missing?

Reply:
This is actually a fix to a bug that is introduced by the upgrade to Timex 3.6 (if I recall correctly). We must retain the `now` value _before_ we called `next_available_date_time` otherwise we end up with `from > until` which now fails.

I should add a comment about this!

Reply:
The code was duplicated, so I extracted it and I added a comment explaining the reason!